lib/Models/ImageryLayerCatalogItem.js

```diff
@@ -705,9 +690,98 @@ ImageryLayerCatalogItem.enableLayer = function(catalogItem, imageryProvider, opa
+                                     getUrlForImageryTile(imageryProvider, tileProviderError.x, tileProviderError.y, tileProviderError.level) + '\n' +
+                                     'succeeded when loaded with XMLHttpRequest but failed multiple times when loaded directly by Cesium or Leaflet.\n' +
+                                     'This may indicate that the server is very sensitive to the request headers provided or is simply unreliable.'
+                        });
+                    } else {
+                        tellMapToRetry();
+                    }
+                }).otherwise(function(e) {
+                    // This attempt failed. We'll either retry or give up depending on the status code.
+                    e = e || {};
+
+                    // We're only concerned about failures for tiles that actually overlap this item's extent.
+                    if (defined(catalogItem.rectangle)) {
+                        var tilingScheme = imageryProvider.tilingScheme;
+                        var tileExtent = tilingScheme.tileXYToRectangle(tileProviderError.x, tileProviderError.y, tileProviderError.level);
+                        var intersection = Rectangle.intersection(tileExtent, catalogItem.rectangle);
+                        if (!defined(intersection)) {
+                            tellMapToSilentlyGiveUp();
+                            return;
+                        }
+                    }
+
+                    if (e.statusCode >= 400 && e.statusCode < 500) {
+                        if (catalogItem.treat404AsError === false && e.statusCode === 404) {
+                            tellMapToSilentlyGiveUp();
+                        } else if (catalogItem.treat403AsError === false && e.statusCode === 403) {
+                            tellMapToSilentlyGiveUp();
+                        } else {
+                            failTile(e);
+                        }
+                    } else if (e.statusCode >= 500 && e.statusCode < 600) {
+                        retryWithBackoff(e);
+                    } else if (!defined(e.statusCode) && defined(e.target)) {
+                        // This is a failed image element, which means we got a 200 response but
+                        // could not load it as an image.
+                        failTile({
+                            name: 'Tile Error',
+                            message: 'Got a 200 (OK) response from URL:\n' +
+                                     getUrlForImageryTile(imageryProvider, tileProviderError.x, tileProviderError.y, tileProviderError.level) + '\n' +
+                                     'but it could not be loaded as an image.'
+                        });
+                    } else if (!defined(e.statusCode)) {
+                        // On a modern-ish browser, this is almost certainly a CORS problem.
+                        // Note that ignoreUnknownTileErrors is only for genuinely unknown (no status code) errors.
```

Comment:
This situation also arises on DNS failures (eg, non existent domain)

Reply:
Good point!